components/vars/tasks/updateDNA.js

```diff
@@ -0,0 +1,565 @@
+          elementVariables[elementName][fullName] = value;
+        }
+        dimensionVariables[fullName] = { name: scaleName, value: value, cssVariableName: cssVariableName };
+        jsElementVariables[elementName][varName] = value;
+      }
+
+      for (let stopName of stops) {
+        let stop = dnaData.elements[stopName];
+
+        for (let scaleName of scales) {
+          let scale = stop[scaleName];
+
+          for (let elementName of elements) {
+            let element = scale[elementName];
+
+            for (let variantName in element) {
+              let variant = element[variantName];
+
+              let metadataKeyBase = 'spectrum-' + elementName + (variantName === 'default' ? '' : `-${variantName}`);
+              metadata[`${metadataKeyBase}-name`] = variant.name;
+              metadata[`${metadataKeyBase}-description`] = variant.description;
+              metadata[`${metadataKeyBase}-status`] = variant.status;
+              metadata[`${metadataKeyBase}-version`] = variant.version;
+
+              if (variant.states) {
+                for (let stateName in variant.states) {
+                  let state = variant.states[stateName];
+
+                  for (let key in state) {
+                    let value = state[key];
+                    let varName = key;
+                    if (isColorValue(varName, value)) {
+                      addColorVariable(elementName, varName, value, variant.varBaseName, stopName, stateName);
+                    }
+                    else {
+                      addDimensionVariable(elementName, varName, value, variant.varBaseName, scaleName, stateName);
+                    }
+                  }
+                }
+              }
+
+              if (variant.colors) {
+                for (let key in variant.colors) {
+                  let value = variant.colors[key];
+                  let varName = key;
+                  addColorVariable(elementName, varName, value, variant.varBaseName, stopName);
+                }
+              }
+
+              if (variant.dimensions) {
+                for (let key in variant.dimensions) {
+                  let value = variant.dimensions[key];
+                  let varName = key;
+                  addDimensionVariable(elementName, varName, value, variant.varBaseName, scaleName);
+                }
+              }
+
+              if (variant.animation) {
+                for (let key in variant.animation) {
+                  let value = variant.animation[key];
+                  let varName = key;
+                  addDimensionVariable(elementName, varName, value, variant.varBaseName, scaleName);
+                }
+              }
+            }
+          }
+        }
+      }
+
+      for (let elementName of elements) {
+        generateCSSFile([
+          elementVariables[elementName]
+        ], elementName, 'css/components');
+        cssFilesGenerated[elementName] = true;
+
+        // generateJSFile([
```

Comment:
maybe remove or add as an optional parameter if the old files should for any reason be compiled.

Reply:
I mentioned in the PR description I've commented them out for now since reviewing a JS API is outside of the scope of this PR. I'd rather not put them behind a flag or anything, just leave the code intact so we can add it back later. I don't think it's a big problem.